c

```diff
@@ -0,0 +1,259 @@
+	scanf("%f", &radnik->koeficijenat);
+	printf("Unesi novi radni staz radnika: \n");
+	scanf("%d", &radnik->radniStaz);
+
+}
+
+bool trazeniKoeficijenat(tipRadnika radnik, float trazeniKoef) {
+	return radnik.koeficijenat == trazeniKoef;
+}
+
+float izracunajTrenutnuZaradu(tipRadnika radnik, float trenutnaCenaRada) {
+	return radnik.koeficijenat * trenutnaCenaRada * (1 + radnik.radniStaz * 0.5 / 100);
+}
+
+int prebrojRadnikeSaTrazenimKoeficijentom(tipRadnika radnici[], int n, float trazeniKoef) {
+	int brojac = 0;
+	for (int i = 0; i < n; i++) {
+		if (radnici[i].koeficijenat == trazeniKoef)
+			brojac++;
+	}
+	return brojac;
+}
+
+int main()
+{
+
+	FILE* pr;
+	int brRadnika, izbor, redniBroj;
+	tipRadnika radnici[100];
+	char imeFajla[80];
+	int i, odaberiSifru, trazenaSifra;
+	float trazeniKoef, trenutnaCenaRada, trenutnaZarada;
+
+
+
+
+	while (1) {
+		puts("****************************************************");
+		puts("* MENI ");
+		puts("* 1 - Otvaranje tekstualnog fajla");
+		puts("* 2 - Snimanje radnika u tekstualni fajl");
+		puts("* 3 - Unos novog radnika");
+		puts("* 4 - Prikaz radnika(sa unetog rednog broja)");
+		puts("* 5 - Prikaz svih radnika");
+		puts("* 6 - Brisanje podataka o nekom radniku");
+		puts("* 7 - Izmena podataka o nekom radniku");
+		puts("* 8 - Provera da li radnik sa datim rednim brojem ima trazeni koeficijenat");
+		puts("* 9 - Izracunavanje zarade radnika sa datim rednim brojem");
+		puts("* 10 - Prebrojavanje radnika sa datim koeficijentom");
+		puts("* 0 - Kraj programa");
+
+
+		printf("Odaberi: \n");
+		scanf("%d", &izbor);
+
+		switch (izbor)
+		{
+
+		case 0:
+			//kraj fajla
+			pr = fopen("radnici.txt", "r");
+			if (pr != NULL)
+				return 0;
+			fclose(pr);
+			break;
+		case 1:
+			// 1 - Otvaranje tekstualnog fajla
+			pr = fopen("radnici.txt", "r");
+			if (pr == NULL)
+				puts("Greska pri otvaranju fajla!");
+			else
+			{
+				i = 0;
+				while (!feof(pr))
+				{
+					procitajRadnike(pr, &radnici[i]);
+					i++;
+				}
+				int n = i;
+				fclose(pr);
+				printf("Fajl je otvoren, procitano je %d radnika!\n", n / 2);
+			}
+			break;
+		case 2:
+			// snimanje radnika u fajl
+			pr = fopen("radnici.txt", "w");
+			if (pr == NULL)
+				puts("Greska pri kreiranju fajla!");
+			else
+			{
+				for (i = 0; i < brRadnika; i++)
+					kopiraj_u_fajl(pr, radnici[i]);
+				fclose(pr);
+				puts("Fajl je kreiran!");
+			}
+			break;
+		case 3:
+			//unos novog radnika
+			printf("Koliko radnika unosis: \n");
+			scanf("%d", &brRadnika);
+			for (i = 0; i < brRadnika; i++) {
+				unosRadnika(&radnici[i]);
+			}
+			break;
+		case 4:
+			//prikaz radnika sa rednim brojem
+			printf("Unesi redni broj radnika za prikaz:\n");
+			scanf("%d", &redniBroj);
+			prikaziRadnika(radnici[redniBroj - 1]);
+			break;
+		case 5:
+			//prikaz svih radnika
+			for (i = 0; i < brRadnika; i++)
+				if (brRadnika > 0)
+					prikaziRadnika(radnici[i]);
+				else
+					printf("Do sada nije unet ni jedan radnik");
```

Comment:
@p4vl4k4  Trenutna logika radi (ali na malo cudan nacin). Ukoliko nemas broj radnika (tj `brRadnika==0`), poruka `Do sada nije unet ni jedan radnik` ti se nikada nece prikazati. 

Razlog za to je zato sto ti najpre iteriras kroz radnike (u ovom slucaju kroz nula radnika) pa onda ispitujes da li je broj radnika veci od nule. Za nula radnika, for petlja nikada nece izvrsiti kod unutar svog tela, a samim tim i nikada neces videti poruku.

Sta bi se desilo ako bi zamenio mesta sa if i for? Necu ti jos reci resenje. Pokusaj sam da resis, pa ako ti je potreban hint, javi se.

Reply:
Ja i moja logika.... Izmenio sam, a sto je najgore profesor koji nam drzi blok, predaje elektroniku(dele nas u 2 grupe i ja sam se zapao u 'goru') 🤯 i kad je pregledavao nije video to. Danas sam ga pitao za nesto, on mi kaze odavno sam ja to radeo, snadji se 😆

Reply:
"Sta bi se desilo ako bi **zamenio** mesta sa if i for?" - ovako vec ima smisla